dart/dynamics/BodyNode.h

```diff
@@ -613,6 +591,60 @@ class BodyNode :
   template <class NodeType, typename ...Args>
   NodeType* createNode(Args&&... args);
 
+  /// Create an ShapeNode attached to this BodyNode. Pass an
+  /// ShapeNode::Properties argument into this function.
+  template <class ShapeNodeProperties>
+  ShapeNode* createShapeNode(const ShapeNodeProperties& properties);
+
+  /// Create an ShapeNode with the specified name
+  ShapeNode* createShapeNode(
+      const ShapePtr& shape, const std::string& name = "ShapeNode");
+
+  /// Create an ShapeNode with the specified name
+  ShapeNode* createShapeNode(const ShapePtr& shape, const char* name);
+
+  /// Create an ShapeNode with the specified name and addons
+  template <class... Addons>
+  ShapeNode* createShapeNode(const ShapePtr& shape,
+                             const std::string& name = "ShapeNode");
```

Comment:
This is only useful if you want to create the `Addon`, but not set any properties on it. Is this something that we expect to happen frequently?